app/models/manageiq/providers/base_manager/event_catcher/runner.rb

```diff
@@ -84,10 +125,24 @@ def monitor_events
     raise NotImplementedError, _("must be implemented in subclass")
   end
 
-  def process_event(_event)
+  # This method has been refactored to go through two steps, namely filetered? and queue_event.
+  # Therefore every subclass should override filtered? and queue_event
+  #
+  # For historical reason existing providers still directly implement process_event
+  # They should be eventually refactored following the example of VMWare provider.
+  def process_event(event)
+    queue_event(event) unless filtered?(event)
+  end
+
+  def queue_event(_event)
     raise NotImplementedError, _("must be implemented in subclass")
```

Comment:
I dont think we should translate these Exceptions, as they don't show up in the UI.
Although it's not part of this PR, maybe remove them in one go?

If not, no worries, and I'll do a PR :)